code/modules/reagents/chemistry/recipes/pyrotechnics.dm

```diff
@@ -116,6 +116,31 @@
 	empulse(location, round(created_volume / 12), round(created_volume / 7), 1)
 	holder.clear_reagents()
 
+
+/datum/chemical_reaction/beesplosion
+	name = "Bee Explosion"
+	id = "beesplosion"
+	required_reagents = list("honey" = 1, "strange_reagent" = 1, "radium" = 1)
+
+/datum/chemical_reaction/beesplosion/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
+	if(created_volume < 5)
+		playsound(location,'sound/effects/sparks1.ogg', 100, 1)
+	else
+		playsound(location,'sound/creatures/bee.ogg', 100, 1)
+		var/list/beeagents = list()
+		for(var/X in holder.reagent_list)
+			var/datum/reagent/R = X
+			if(R.id in required_reagents)
+				continue
+			beeagents += R
+		var/bee_amount = round(created_volume/5)
```

Comment:
`round(created_volume * 0.2)` will be slightly faster than `round(created_volume / 5)`

Reply:
this is also required by contributing.md